### PR TITLE
[R] An easy-to-use interface for mlp training

### PR DESCRIPTION
**Pull request by hetong007**
For mlp, the symbolic connection can be built automatically. The "5 minutes" tutorials will be updated with this interface soon.

### CODE REVIEW

File: R-package/R/mlp.R [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    act <- mx.symbol.Activation(fc, act_type=activation[i])
+    if (!is.null(dropout)) {
+      act <- mx.symbol.Dropout(act, p = dropout[i+1])
+    }
+  }
+  fc <- mx.symbol.FullyConnected(act, num_hidden=out_node)
+  if (out_activation == "rmse") {
+    out <- mx.symbol.LinearRegressionOutput(fc)
+  } else if (out_activation == "softmax") {
+    out <- mx.symbol.Softmax(fc)
```
Comment (tqchen) [inline R-package/R/mlp.R]: use SoftmaxOutput instead of Softmax, c.f. #434

File: R-package/R/mlp.R [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#' preds = predict(model, test.x)
+#' 
+#' @export
+mx.mlp <- function(data, label, hidden_node = 1, out_node, dropout = NULL, 
+                   activation = "tanh", out_activation = "softmax",
+                   device=mx.ctx.default(), ...) {
+  
+  m <- length(hidden_node)
+  if (!is.null(dropout)) {
+    if (length(dropout) == 1) {
```
Comment (tqchen) [inline R-package/R/mlp.R]: dropout is usually only applied to last layer by default

Comment (hetong007) [inline R-package/R/mlp.R]: Do you here mean in last hidden -> output? what about a denoising autoencoder?

File: R-package/R/mlp.R (hunk 1 above)
Comment (hetong007) [inline R-package/R/mlp.R]: Seems SoftmaxOutput hasn't been defined yet, I got "could not find function "mx.symbol.SoftmaxOutput""

File: R-package/R/mlp.R (hunk 2 above)
Comment (tqchen) [inline R-package/R/mlp.R]: Autoencoder was indeed different enough. For normal mlp, dropout was usually only before the last layer

File: R-package/R/mlp.R (hunk 1 above)
Comment (tqchen) [inline R-package/R/mlp.R]: Need to make the latest mxnet and do make rcppexport
